src/Authoring/WinRT.SourceGenerator/DiagnosticUtils.cs

```diff
@@ -222,13 +223,18 @@ private void CheckMethods(IEnumerable<MethodDeclarationSyntax> methodDeclaration
         /// <summary>Looks at all the properties of the given class and checks them for improper array types (System.Array instances, multidimensional, jagged)</summary>
         ///<param name="props">collection of properties</param><param name="typeId">containing class/interface</param>
         /// <returns>True iff any of the invalid array types are used in any of the propertyy signatures in the given class</returns>
-        private void CheckPropertySignature(IEnumerable<PropertyDeclarationSyntax> props, SyntaxToken typeId)
+        private void CheckProperties(IEnumerable<PropertyDeclarationSyntax> props, SyntaxToken typeId)
         {
             foreach (var prop in props)
             {
                 var propSym = GetModel(prop.SyntaxTree).GetDeclaredSymbol(prop);
                 var loc = prop.GetLocation();
 
+                if (!propSym.GetMethod.DeclaredAccessibility.Equals(Accessibility.Public))

```

Comment:
Does this error if they didn't have a get method at all?  We probably want to catch that case too.

Reply:
No it doesn't. I added a unittest for the scenario you are describing and it passes with no other changes. I think this is because all methods are considered public unless otherwise declared